NpmPS/Public/Publish-NpmPackage.ps1

```diff
@@ -0,0 +1,192 @@
+function Publish-NpmPackage
+{
+    <#
+        .Synopsis
+        Publishes a npm package
+
+        .Example
+
+        $registry = 'https://contoso.local/npm'
+        $credential = Get-Credential
+
+        $publishLDNpmPackageSplat = @{
+            Credential = $credential
+            Path       = $path
+            Registry   = $registry
+            Version    = '0.1.0-rc.1'
+            Tag        = 'testrelease'
+        }
+        Publish-NpmPackage @publishLDNpmPackageSplat
+
+        .Notes
+
+    #>
+    [cmdletbinding()]
+    param(
+        # Location of the package.json
+        [Parameter(
+            Mandatory,
+            Position = 0,
+            ValueFromPipelineByPropertyName
+        )]
+        [ValidateNotNullOrEmpty()]
+        [String]
+        $Path,
+
+        # NPM Registry to publish
+        [Alias('Repository')]
+        [String]
+        $Registry,
+
+        # Username and API Token as password
+        [PSCredential]
+        $Credential,
+
+        # SemVer
+        [String]
+        $Version,
+
+        # tags to set when publishing
+        [Alias('Tags')]
+        [String[]]
+        $Tag,
+
+        # Force publish even if package already exists
+        [Switch]
+        $Force
+    )
+
+    begin
+    {
+        $npmProtocolRegex = '^http?:'
```

Comment:
should be '^https?' correct? optional s, not optional p?

Reply:
corrected